### PR TITLE
Jetpack Plans: New UI & switch to redux process for the automated setup 

### PR DESCRIPTION
**Pull request by ryelle**
**Copied from #5244, see there for history.**

This adds a much nicer UI for automatically setting up plugins purchased in a wordpress.org plan. After purchasing a plan, the user will be redirected to this page (redirect not set up yet), where we will try to install & set up the plugins they've just bought.

Preview: https://cloudup.com/cLIJ99_Njhu

The most basic test:

- Have a connected Jetpack site with Premium or Business, and no VaultPress, Akismet, or Polldaddy installed.
- Visit http://calypso.localhost:3000/plugins/setup/{site}
- Watch as it installs VaultPress, Akismet, and (if you're Business) Polldaddy _(the plugins installed are controlled by a wp.com API response, and can be configured if plans change)_
- Visit your website's wp-admin, you should see a "Welcome to VaultPress" message, and Akismet & Polldaddy will be (silently) set up.
- If you'd rather check this in your database, you'll see the `vaultpress`, `wordpress_api_key`, and `polldaddy_api_key` are all set with your keys.

cc @johnHackworth @rickybanister

### CODE REVIEW

File: client/my-sites/plugins/plan-setup/index.jsx [hunk 1]
```diff
@@ -1,81 +1,336 @@
 /**
```
Comment (mtias) [inline client/my-sites/plugins/plan-setup/index.jsx]: @johnHackworth this file needs some work on the namespace and classes (`plan-setup` is too generic) that don't match the folder namespace.

File: client/my-sites/plugins/plan-setup/style.scss [hunk 2]
```diff
@@ -0,0 +1,54 @@
+$vaultpress-red: #ac0300;
+$akismet-green:  #7C992E;
+$polldaddy-red:  #D45458;
+
+.plan-setup {
```
Comment (mtias) [inline client/my-sites/plugins/plan-setup/style.scss]: plan-setup is too generic, is this only supposed to work for jetpack?

File: client/my-sites/plugins/plan-setup/style.scss [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+.plan-setup__header {
+	@extend %heading;
+	margin-bottom: .75em;
+	text-align: center;
+	color: $gray-dark;
+}
+
+.plan-setup__description {
+	margin-bottom: 3em;
```
Comment (mtias) [inline client/my-sites/plugins/plan-setup/style.scss]: We favor px values in general.

File: client/my-sites/plugins/plan-setup/index.jsx (hunk 1 above)
Comment (ryelle) [inline client/my-sites/plugins/plan-setup/index.jsx]: @mtias sure, `plan-setup` is pretty generic. would `plugins-setup` still be too generic? `plugins-jetpack-setup`? trying to follow the convention of the neighbor folders with the `plugins-` prefix.

Comment (mtias) [inline client/my-sites/plugins/plan-setup/index.jsx]: Does plugins make the most sense to describe this? `jetpack-setup`? I'm not familiar with what is being setup :)

Comment (ryelle) [inline client/my-sites/plugins/plan-setup/index.jsx]: The plugins you've just purchased as part of your jetpack plan (VaultPress/Akismet/Polldaddy). Not Jetpack itself, that's already set up if you've made it this far. `jetpack-plan-setup` might also make sense, `jetpack-plugins-setup` if the prefix is not important.

Comment (mtias) [inline client/my-sites/plugins/plan-setup/index.jsx]: Got it. Whatever makes the most sense to you, just having the jetpack prefix fixes the main problem.

**Comment by johnHackworth** [comment]
I think this is ready to merge too :ship:

**Pull request merged**